packages/nimble-components/src/rich-text-editor/tests/rich-text-editor.spec.ts

```diff
@@ -447,4 +447,662 @@ describe('RichTextEditor', () => {
             );
         });
     });
+
+    it('setting an empty string should clear a value in the editor', () => {
+        element.setMarkdown('markdown string');
+        expect(pageObject.getEditorTagNames()).toEqual(['P']);
+        expect(pageObject.getEditorLeafContents()).toEqual(['markdown string']);
+
+        element.setMarkdown('');
+        expect(pageObject.getEditorTagNames()).toEqual(['P', 'BR']);
+        expect(pageObject.getEditorLeafContents()).toEqual(['']);
+
+        element.setMarkdown('new markdown string');
+        expect(pageObject.getEditorTagNames()).toEqual(['P']);
+        expect(pageObject.getEditorLeafContents()).toEqual([
+            'new markdown string'
+        ]);
+    });
+
+    describe('supported rich text formatting options from markdown string to its respective HTML elements', () => {
```

Comment:
Are these test cases identical to the ones that already verify the markdown to DOM conversion for the rich text viewer? I mostly want to make sure that our coverage is good for both components.

We could also think about ways to share the tests. For example, share a data structure with markdown values and expected tags / leaf contents then have both components test suites iterate over that data. But I'm also OK with duplicating the tests as long as we're diligent about keeping them in sync.

Reply:
As mentioned [here](https://github.com/ni/nimble/pull/1424#discussion_r1291010057), refactoring PR parsing logic between viewer and editor and moving it to the same files under `models` is in the pipeline.